lib/parser/livesplit_core_parser.rb

```diff
@@ -0,0 +1,107 @@
+require File.expand_path("../LiveSplitCore", __FILE__)
+# bindings compiled from https://github.com/CryZe/livesplit-core master@da78705
+# so file  compiled from https://github.com/CryZe/livesplit-core master@da78705
+
+class Parser
+  def self.parse(run_string, fast: false)
+    parse_result = LiveSplitCore::Run.parse(run_string, run_string.bytesize)
+    if !parse_result.parsed_successfully
+      return nil
+    end
+    program = parse_result.timer_kind
+    run = parse_result.unwrap()
+
+    run_object = {
+      program: program,
+      game: run.game_name(),
+      category: run.category_name(),
+      srdc_id: run.metadata().run_id(),
+      attempts: run.attempt_count(),
+      offset: run.offset().total_seconds(),
+      history: nil,
+      indexed_history: nil,
+      realtime_history: nil,
+      splits: [],
+      realtime_time: 0,
+      gametime_time: 0
+    }.tap { |run| run[:name] = "#{run[:game]} #{run[:category]}".strip }
+
+    unless fast
+      run_object[:history] = []
+      run_object[:indexed_history] = {}
+      run_object[:realtime_history] = []
+      (0...run.attempt_history_len()).each do |i|
+        attempt = run.attempt_history_index(i)
+        attempt_id = attempt.index().to_i
+        time = attempt.time()
+
+        run_object[:history][attempt_id] = {
+          gametime: time.game_time().try(:total_seconds) || 0,
+          realtime: time.real_time().try(:total_seconds) || 0
+        }
```

Comment:
if nothing is using this new history field yet, can we put milliseconds in it instead? that way when we start using it later we don't have to convert anything.

Reply:
Totally, should I also change the new split.history hash to also convert down to milliseconds?

Reply:
yeah, if it's not an unreasonable amount of work

Reply:
Turns out indexed_history is used by other methods, and I would have to add a new field to the db to achieve this which seems a bit out of scope for this PR.